### ISSUE TITLE
Tacotron2 is generating new mel spectograms and alignment for the same input text sequence while inferencing.

### ISSUE THREAD

**Issue by nav99een** [NVIDIA/tacotron2#553]
Hi everyone,
I am trying to generate speech from text using tacotron2 finetuned on my custom dataset. So, while inferencing, As you can see from the image attached below, I am getting mel-spectograms with different shape and alignments for the same input sequence.

![issue](https://user-images.githubusercontent.com/42210253/158355573-b8c05a33-a58b-47f3-81b2-8dbe43429343.jpg)

 I tried removing the dropout layer in prenet but then model generates no alignment at all. I tried almost all other suggested approaches for the same but it failed. I also tried to perform inference using torch.no_grad(). I don't want to generate new alignments everytime. 
Please help me out. Thanks in advance!!

**Comment by PiotrDabkowski** [comment]
This is due to the prenet dropout, the problem is that the network is now dependent on the prenet dropout and does not generalize to it being off. The tacotron2 is hence not deterministic by design (at lest as implemented in this repo).

**Comment by lirus7** [comment]
@PiotrDabkowski, Any specific reason for pre-net dropout being different from a regular dropout?

**Comment by PiotrDabkowski** [comment]
It disrupts feedback. In mel spec the frames are highly similar, you can just predict the previous frame to do quite well on the loss fn.  If you remove the dropout then this can lead to positive feedback. I guess the solution here would be to randomly turn off prenet droput during training. 

Also you can use manual seed during inference to get the same results.

**Comment by Sytronik** [comment]
Here are useful insights about dropout in PreNet: https://github.com/mozilla/TTS/issues/50
